registry/storage/driver/s3/s3.go

```diff
@@ -587,6 +587,11 @@ func (d *driver) List(path string) ([]string, error) {
 	if path != "/" && path[len(path)-1] != '/' {
 		path = path + "/"
 	}
+	prefix := ""
```

Comment:
I fundamentally distrust random blocks of code like this. A little helper function can add some much needed context for future understanding. Mind you, this is more about communicating the intent of the code, rather than what it does. A comment would be sufficient.